src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs

```diff
@@ -356,47 +356,102 @@ public static uint[] ComputeBytesToUint(ReadOnlySpan<byte> input, int size)
         // compute a Keccak hash (md) of given byte length from "in"
         public static void ComputeHash(ReadOnlySpan<byte> input, Span<byte> output)
         {
-            int size = output.Length;
-            int roundSize = GetRoundSize(size);
+            int roundSize = GetRoundSize(output.Length);
             if (output.Length <= 0 || output.Length > STATE_SIZE)
             {
                 ThrowBadKeccak();
             }
 
             Span<ulong> state = stackalloc ulong[STATE_SIZE / sizeof(ulong)];
-            Span<byte> temp = stackalloc byte[TEMP_BUFF_SIZE];
+            Span<byte> stateBytes = MemoryMarshal.AsBytes(state);
 
-            int remainingInputLength = input.Length;
-            for (; remainingInputLength >= roundSize; remainingInputLength -= roundSize, input = input[roundSize..])
-            {
-                ReadOnlySpan<ulong> input64 = MemoryMarshal.Cast<byte, ulong>(input[..roundSize]);
-
-                for (int i = 0; i < input64.Length; i++)
-                {
-                    state[i] ^= input64[i];
-                }
+            int inputLength = input.Length;
+            int offset = 0;
 
+            // Process full rounds
+            while (inputLength - offset >= roundSize)
+            {
+                XorVectors(stateBytes, input.Slice(offset, roundSize));
                 KeccakF(state);
+                offset += roundSize;
             }
 
-            // last block and padding
-            if (input.Length >= TEMP_BUFF_SIZE || input.Length > roundSize || roundSize + 1 >= TEMP_BUFF_SIZE || roundSize == 0 || roundSize - 1 >= TEMP_BUFF_SIZE)
+            int remainingInputLength = inputLength - offset;
+            if (remainingInputLength > 0)
             {
-                ThrowBadKeccak();
+                // XOR the remaining input bytes into the state
+                XorVectors(stateBytes, input.Slice(offset, remainingInputLength));
             }
 
-            input[..remainingInputLength].CopyTo(temp);
-            temp[remainingInputLength] = 1;
-            temp[roundSize - 1] |= 0x80;
-
-            Span<ulong> tempU64 = MemoryMarshal.Cast<byte, ulong>(temp[..roundSize]);
-            for (int i = 0; i < tempU64.Length; i++)
+            // Apply padding
+            if (remainingInputLength != roundSize)
             {
-                state[i] ^= tempU64[i];
+                // Apply padding within the current block
+                stateBytes[remainingInputLength] ^= 0x01; // Append bit '1' after the input
+                stateBytes[roundSize - 1] ^= 0x80;        // Set the last bit of the block to '1'
             }
 
+            // Process the final block
             KeccakF(state);
-            MemoryMarshal.AsBytes(state[..(size / sizeof(ulong))]).CopyTo(output);
+
+            if (output.Length == Vector256<byte>.Count)
```

Comment:
no `Vector128`?

Reply:
Not for copying out as hashes are only 256 and 512